### ISSUE TITLE
gcc 10/clang-11 + i686: sse/mm_or_ps avx512/{andnot/mm512_mask_andnot_ps,xor/mm512_xor_ps} test failure

### ISSUE THREAD

**Issue by mr-c** [simd-everywhere/simde#683]
SIMDe 0.7.0 on gcc-10 (10.2.1-1) and clang-11 (1:11.0.0-5+b1) (both from Debian Testing)

https://ci.debian.net/data/autopkgtest/testing/i386/s/simde/9192262/log.gz

```
/usr/bin/gcc -DSIMDE_BUILD_CPP_TESTS -I/tmp/autopkgtest-lxc._j_861et/downtmp/autopkgtest_tmp/test/.. -DSIMDE_ENABLE_OPENMP -fopenmp-simd -g -std=gnu99 -Wno-psabi -Wno-psabi -o CMakeFiles/simde-test-native.dir/x86/sse.c.o -c /tmp/autopkgtest-lxc._j_861et/downtmp/autopkgtest_tmp/test/x86/sse.c
/x86/sse/mm_or_ps/native/c           [ ERROR ]
/tmp/autopkgtest-lxc._j_861et/downtmp/autopkgtest_tmp/test/x86/sse.c:4343: assertion failed: r[3] == simde_x_mm_loadu_epi32(test_vec[i].r)[3] (2147345310 == 2143151006)

/usr/bin/gcc -DSIMDE_BUILD_CPP_TESTS -I/tmp/autopkgtest-lxc._j_861et/downtmp/autopkgtest_tmp/test/.. -DSIMDE_ENABLE_OPENMP -fopenmp-simd -g -std=gnu99 -Wno-psabi -Wno-psabi -o CMakeFiles/simde-test-native.dir/x86/avx512/andnot.c.o -c /tmp/autopkgtest-lxc._j_861et/downtmp/autopkgtest_tmp/test/x86/avx512/andnot.c
/x86/avx512/andnot/mm512_mask_andnot_ps/native/c[ ERROR ]
/tmp/autopkgtest-lxc._j_861et/downtmp/autopkgtest_tmp/test/x86/avx512/andnot.c:409: assertion failed: r[5] == test_vec[i].r[5] (-3917627 == -8111931)

/usr/bin/gcc -DSIMDE_BUILD_CPP_TESTS -I/tmp/autopkgtest-lxc._j_861et/downtmp/autopkgtest_tmp/test/.. -DSIMDE_ENABLE_OPENMP -fopenmp-simd -g -std=gnu99 -Wno-psabi -Wno-psabi -o CMakeFiles/simde-test-native.dir/x86/avx512/xor.c.o -c /tmp/autopkgtest-lxc._j_861et/downtmp/autopkgtest_tmp/test/x86/avx512/xor.c
/x86/avx512/xor/mm512_xor_ps/native/c[ ERROR ]
/tmp/autopkgtest-lxc._j_861et/downtmp/autopkgtest_tmp/test/x86/avx512/xor.c:96: assertion failed: r[0] == simde_mm512_loadu_epi32(test_vec[i].r)[0] (1333546396 == 1329352092)

**Comment by mr-c** [comment]
On Debian unstable the gcc error is gone ([gcc version 10.2.1-3](https://tracker.debian.org/media/packages/g/gcc-10/changelog-10.2.1-3)), but even with the newer clang version ([1:11.0.1~+rc2-1](https://tracker.debian.org/media/packages/l/llvm-toolchain-11/changelog-111.0.1rc2-1)) clang still fails the same tests as aboe

**Comment by mr-c** [comment]
This appears to have been fixed in clang 11.0.1-2 + latest SIMDe
